_source/_posts/2020-10-04-introducing-okta-cli.md

```diff
@@ -0,0 +1,319 @@
+
+```properties
+okta.oauth2.issuer=${CLI_OKTA_ISSUER}
+okta.oauth2.client-id=${CLI_OKTA_CLIENT_ID}
+okta.oauth2.client-secret=${CLI_OKTA_CLIENT_SECRET}
+```
+
+3. Create a metadata file in your project at: `.okta/sample-config.yaml`:
+
+```yaml
+oauthClient:
+  redirectUris:
+    - http://localhost:8080/login/oauth2/code/okta
+  applicationType: web
+```
+
+4. Run `okta start`:
+
+```bash
+Configuring a new OIDC Application, almost done:
+Created OIDC application, client-id: 0oa6mv2tHQ1IPbG2r5d5
+```
+
+You can see that the placeholders in your `src/main/resources/application.properties` file have been replaced:
+
+```bash
+cat src/main/resources/application.properties
+
+okta.oauth2.issuer=https://dev-159750.okta.com/oauth2/default
+okta.oauth2.client-id=0oa6mv2tHQ1IPbG2r5d5
+okta.oauth2.client-secret=tgIdHYX4EvRUlt0HnuRV-y_9qlr9C1IM4C9Z-g17
+```
+
+At this point, you could fire up your app as normal and have it fully integrated with Okta for authentication:
+
+```bash
+./mvnw spring-boot:run
+```
+
+The important bits to set up a project for use with the Okta CLI are:
+
+1. Add the `OKTA_CLI_` placeholder values to the configuration files that make sense for your language and framework
+2. Add the `.okta/sample-config.yaml` file with the right application type for your language and framework
+3. Run `okta start` to provision and Okta app and update all the placeholders in your config file(s).
+
+> The easiest way to see what configuration you need for your language and framework is to visit the official samples found in the [okta-samples](https://github.com/okta-samples) GitHub org.
+
+Currently, the available placeholder values are:
+
+| Placeholder            | Description                                                                                              |
+|------------------------|----------------------------------------------------------------------------------------------------------|
+| CLI_OKTA_ORG_URL       | The base URL to your Okta Org                                                                            |
+| CLI_OKTA_ISSUER        | The URL to your Okta OAuth 2.0 [Authorization Server](https://developer.okta.com/docs/concepts/auth-servers/) |
+| CLI_OKTA_CLIENT_ID     | The client id set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                 |
+| CLI_OKTA_CLIENT_SECRET | The client secret set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                              |
+
+## Adding Samples for Okta CLI
```

Comment:
Unfortunately, it's probably best to remove this section. Unless you want to offer an OktaDev t-shirt for creating a PR.